src/model.rs

```diff
@@ -27,6 +27,7 @@ pub struct Model {
     env: Env,
     pub(crate) vars: IdxManager<Var>,
     pub(crate) constrs: IdxManager<Constr>,
+    pub(crate) genconstrs: IdxManager<GenConstr>,
```

Comment:
> have the methods/FFI functions for adding general constraints been added yet?
I see this PR as more of a basis. There are multiple types of general constraints, and I suspect that adding each one would be non-trivial, so I'd make a PR for each one.
I've actually implement indicator constraints (partially because they are the ones I originally needed :sweat_smile:), and thought I'd make a PR for them after this one is merged

Reply:
3-SAT: addressed in 8b9c652d676b023fe265705dcbc6f2ab7e6b7d93
PWL: addressed in de86c26dcc4c5ec94f38710ca5c5e46636b0fcc5

Reply:
The PWL test is kind of slow by the way (at least on my laptop). It's most likely because the test shows both the manual implementation of the "function constraint -> PWL constraint" (which unsurprisingly turns out to be slow and inefficient), and Gurobi's automatic conversion (which is quite snappy indeed). Maybe move it from tests to examples? Or somehow mock the first half while still running the second?